pkg/webhook/webhook.go

```diff
@@ -74,6 +82,35 @@ func AddToManager(mgr manager.Manager, poolManager *pool_manager.PoolManager) er
 	if err != nil {
 		return err
 	}
-
 	return nil
 }
+
+func CreateOwnerRefForMutatingWebhook(kubeClient *kubernetes.Clientset) error {
+	managerStatefulSet, err := kubeClient.AppsV1().StatefulSets("kubemacpool-system").Get("kubemacpool-mac-controller-manager", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	ownerRefList := []metav1.OwnerReference{{Name: managerStatefulSet.Name, Kind: "StatefulSet", APIVersion: "apps/v1", UID: managerStatefulSet.UID}}
+
+	mutatingWebHookObject, err := kubeClient.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Get("kubemacpool", metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
why do we create it ourselves? it is already being created, isn't it? this looks like a duplicate.

Reply:
see the comment below

Reply:
Referencing Service was one thing. In this comment I was talking about the fact, that MutatingWebhookConfiguration is dynamically created by kubemacpool even without this PR (am I right?). I don't see you removing that code in this PR, so we have a duplicate.

I wonder whether we could edit the code that creates mentioned MutatingWebhookConfiguration, so it would set reference before creating.

Reply:
the only way to do it is move the code from the vendor folder and make it our code but I don't like this solution (bad for upgrade the packages).

There is no problem on creating the webhook first is the same as if the manager restart itself.
The code check the related spec variables he need but I doesn't change the `ObjectMeta`